packages/amplify-provider-awscloudformation/src/utils/admin-login-server.ts

```diff
@@ -0,0 +1,127 @@
+import { stateManager } from 'amplify-cli-core';
+import { CognitoIdentity } from 'aws-sdk';
+import bodyParser from 'body-parser';
+import cors from 'cors';
+import express from 'express';
+import { JWK, JWKS, JWT } from 'jose';
+import _ from 'lodash';
+import fetch from 'node-fetch';
+
+import { TokenPayload, CognitoIdToken, CognitoAccessToken } from './cognitoJwtTypes';
+
+export class AdminLoginServer {
+  private port = 4242; // placeholder
+  private app: express.Application;
+  private server;
+  private appId: string;
+
+  private corsOptions: {
+    origin: string[];
+    methods: string[];
+    allowedHeaders: string;
+  };
+
+  constructor(appId: string, originUrl: string) {
+    this.appId = appId;
+    this.corsOptions = {
+      origin: [originUrl],
+      methods: ['POST', 'OPTIONS'],
+      allowedHeaders: 'Content-Type',
+    };
+    this.app = express();
+    this.app.use(cors(this.corsOptions));
+    this.app.use(bodyParser.json());
+  }
+
+  public async startServer(callback: () => void) {
+    await this.setupRoute(callback);
+    this.server = this.app.listen(this.getPort());
+  }
+
+  // TODO: scan for available ports across a range like mock
```

Comment:
Yes.

Reply:
This over-ompitization at this point. We have a fixed port for amplify mock and we should do something similar her as well until customers ask for it or we run into issue. Can leave it as todo.